packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -1057,7 +1057,11 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
       final int completedCycles = _framesEmitted ~/ _codec!.frameCount;
       if (_codec!.repetitionCount == -1 || completedCycles <= _codec!.repetitionCount) {
         _decodeNextFrameAndSchedule();
+        return;
       }
+
+      _codec!.dispose();
```

Comment:
We should document on the `MultiFrameImageStreamCompleter` that it takes over ownership of the Codec that was passed in and will dispose it if it is done with it.